airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobCreator.java

```diff
@@ -38,22 +39,29 @@
 import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
 import io.airbyte.protocol.models.DestinationSyncMode;
 import io.airbyte.protocol.models.SyncMode;
+import io.airbyte.scheduler.persistence.job_factory.OAuthConfigSupplier;
 import java.io.IOException;
 import java.util.List;
 import java.util.Optional;
 
 public class DefaultJobCreator implements JobCreator {
 
   private final JobPersistence jobPersistence;
+  private final OAuthConfigSupplier oAuthConfigSupplier;
 
-  public DefaultJobCreator(JobPersistence jobPersistence) {
+  public DefaultJobCreator(JobPersistence jobPersistence, OAuthConfigSupplier oAuthConfigSupplier) {
     this.jobPersistence = jobPersistence;
+    this.oAuthConfigSupplier = oAuthConfigSupplier;
   }
 
   @Override
   public long createSourceCheckConnectionJob(SourceConnection source, String dockerImageName) throws IOException {
+    final JsonNode sourceConfiguration = oAuthConfigSupplier.injectSourceOAuthParameters(
```

Comment:
The combining for get spec / check / discover can't happen in the JobCreator since the jobs are created directly in the `DefaultSynchronousSchedulerClient`. We're removing these misleading and unused functions in https://github.com/airbytehq/airbyte/pull/5896

Reply:
Oh oh, so I am reverting back to injecting OAuth parameters in the `DefaultSyncJobFactory` as well as handling it in `DefaultSynchronousSchedulerClient` for check/discover. (probably not needed for get spec)